gnovm/pkg/gnolang/realm.go

```diff
@@ -220,9 +264,127 @@ func (rlm *Realm) DidUpdate(po, xo, co Object) {
 	}
 }
 
+// checkCrossRealm2 checks cross realm recursively.
+func checkCrossRealm2(rlm *Realm, store Store, tv *TypedValue, isLastRef bool, seenObjs []Object) {
```

Comment:
do we have tests checking for some kinds of circular references?

Reply:
please see zrealm_crossrealm20c3.gno

Reply:
```go
// PKGPATH: gno.land/r/crossrealm_test
package crossrealm_test

type Node struct {
	*Node
}

var root interface{}

func main() {
	node := &Node{}
	node.Node = node

	root = node
	println("ok")
}

// Output:
// ok
```